source/distributed/fully_distributed_tria.cc

```diff
@@ -539,35 +540,39 @@ namespace parallel
         dealii::Utilities::pack(construction_data, buffer, false);
 
         // Write offsets to file.
-        unsigned int buffer_size = buffer.size();
+        const std::uint64_t buffer_size = buffer.size();
 
-        unsigned int offset = 0;
+        std::uint64_t offset = 0;
 
         ierr = MPI_Exscan(&buffer_size,
                           &offset,
                           1,
-                          MPI_UNSIGNED,
+                          MPI_UINT64_T,
                           MPI_SUM,
                           this->mpi_communicator);
         AssertThrowMPI(ierr);
 
         // Write offsets to file.
         ierr = MPI_File_write_at(fh,
-                                 myrank * sizeof(unsigned int),
+                                 myrank * sizeof(std::uint64_t),
                                  &buffer_size,
                                  1,
-                                 MPI_UNSIGNED,
+                                 MPI_UINT64_T,
```

Comment:
A slightly better alternative would have been to use this instead:
```suggestion
                                 Utilities::MPI::mpi_type_id_for_type<decltype(buffer_size)>,
```
to make sure that type and variable remain in sync.